libs/libc/semaphore/sem_init.c

```diff
@@ -78,10 +78,8 @@ int nxsem_init(FAR sem_t *sem, int pshared, unsigned int value)
 #  if CONFIG_SEM_PREALLOCHOLDERS > 0
       sem->hhead            = NULL;
 #  else
-      sem->holder[0].htcb   = NULL;
-      sem->holder[0].counts = 0;
-      sem->holder[1].htcb   = NULL;
-      sem->holder[1].counts = 0;
+      sem->holder[0]        = (struct semholder_s) SEMHOLDER_INITIALIZER;
+      sem->holder[1]        = (struct semholder_s) SEMHOLDER_INITIALIZER;
```

Comment:
I'm not sure if this is allowed with C89. Please check and if it is not supported, then we need to rework this part.

Reply:
Reworked, thanks!